auth/end-user/web/main.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An example web application that obtains authorization and credentials from
+an end user."""
+
+import os
+
+import flask
+import google.auth.transport.requests
+import google.oauth2.credentials
+import google_auth_oauthlib.flow
+
+# The path to the client-secrets.json file obtained from the Google API
+# Console. You must set this before running this application.
+CLIENT_SECRETS_FILENAME = os.environ['GOOGLE_CLIENT_SECRETS']
+# The OAuth 2.0 scopes that this application will ask the user for. In this
+# case, we're asking for basic profile information.
+SCOPES = ['email', 'profile']
+
+app = flask.Flask(__name__)
+# Note: A secret key is included in the sample so that it works, but if you
+# use this code in your application please replace this with a truly secret
+# key. See http://flask.pocoo.org/docs/0.12/quickstart/#sessions.
+app.secret_key = 'example key, please replace.'
+
+
+@app.route('/')
+def index():
+    if 'credentials' not in flask.session:
+        return flask.redirect('authorize')
+
+    # Load the credentials from the session.
+    credentials = google.oauth2.credentials.Credentials(
+        **flask.session['credentials'])
+
+    # Get the basic user info from the Google OAuth2.0 API. This
+    # sample uses the Requests library to make a direct HTTP request to
+    # a Google API, but you can also pass the credentials into the Google
+    # API Client Library and Google Cloud Client Library.
+    session = google.auth.transport.requests.AuthorizedSession(credentials)
+
+    response = session.get('https://www.googleapis.com/oauth2/v2/userinfo')
```

Comment:
You're calling the OAuth2 API in this example. Since this is an example for how to use auth, it might be less completely if the API was something unrelated (like GMail or Maps)

Reply:
It's actual quite lame that this API is called the "oauth2" API. It's a great candidate for this sample because it doesn't require being turned on in the API console, has a stupid high quota, and is extremely deterministic in what it returns.

I could maybe alternatively use the gmail api and fetch the user profile via that, but I'd like to see how this fairs before doing that.